docs/labs/docker-plugin.rst

```diff
@@ -0,0 +1,147 @@
+
+This currently depends on the `experimental build of Docker <https://github.com/docker/docker/blob/master/experimental/>`_.
+
+See also the `GitHub repo for this project <https://github.com/ClusterHQ/flocker-docker-plugin>`_.
+
+.. _labs-architecture-note:
+
+.. note::
+    Note that you should either use the Flocker Docker plugin to associate containers with volumes (the integration architecture described above), or you should use the :ref:`Flocker containers API <api>` and :ref:`flocker-deploy CLI <cli>`, but not both.
+
+    They are distinct architectures.
+    The integration approach allows Docker to control Flocker via the Flocker Dataset API.
+    This allows Flocker to be used in conjunction with other ecosystem tools like :ref:`Docker Swarm <labs-swarm>` and :ref:`Docker Compose <labs-compose>`, which is more experimental than ``flocker-deploy`` and the Flocker containers API.
+
+How it works
+============
+
+The Flocker Docker plugin means you can run containers with named volumes without worrying which server your data is on.
+
+The plugin will create or move the volumes in place as necessary.
+
+The Flocker Docker plugin operates on the ``name`` passed to Docker in the ``docker run`` command and associates it with a Flocker dataset with the same name (i.e. with metadata ``name=foo``).
+
+There are three main cases which the plugin handles:
+
+* If the volume does not exist at all on the Flocker cluster, it is created on the host which requested it.
+* If the volume exists on a different host, it is moved in-place before the container is started.
+* If the volume exists on the current host, the container is started straight away.
+
+Multiple containers can use the same Flocker volume (by referencing the same volume name, or by using Docker's ``--volumes-from``) so long as they are running on the same host.
+
+Demo
+====
+
+This demo shows both the Flocker Docker plugin in conjunction with the :ref:`Volumes CLI <labs-volumes-cli>` and :ref:`Volumes GUI <labs-volumes-gui>`.
+
+.. raw:: html
+
+   <iframe width="100%" height="450" src="https://www.youtube.com/embed/OhWxJ_hOPx8?rel=0&amp;showinfo=0" frameborder="0" allowfullscreen style="margin-top:1em;"></iframe>
+
+Also check out the `DockerCon Plugin Demos <https://plugins-demo-2015.github.io/>`_ site to see a joint project between ClusterHQ and Weaveworks.
+This is the "ultimate integration demo" — a pre-built demo environment that includes Flocker, Weave, Swarm, Compose & Docker, all working together in harmony.
+
+Quickstart installation
+=======================
+
+You can use the ``flocker-plugin-install`` tool which is part of the :ref:`installer <labs-installer>` to quickly install the Flocker Docker Plugin on a cluster you set up with that tool.
+
+Otherwise, if you want to install the Flocker Docker plugin manually, you can follow the following instructions.
+
+Manual Installation on Ubuntu 14.04
+===================================
+
+First install Flocker on some hosts.
```

Comment:
we've a bit of a host v nodes problem in the main docs, in that we use both. I think nodes is winning due to its consistent use throughout the code. It would be helpful if we had a similar approach in the labs docs.